CRM/Utils/Hook.php

```diff
@@ -1635,6 +1635,23 @@ public static function disable() {
     );
   }
 
+  /**
+   * Alter redirect.
+   *
+   * This hook is called when the browser is being re-directed and allows the url
+   * to be altered.
+   *
+   * @param string $url
+   *
+   * @return mixed
+   */
+  public static function alterRedirect($url) {
+    return self::singleton()->invoke(1, $url,
```

Comment:
Use the notation `invoke(array('url'), $url, ...) to fully support all styles of listeners (e.g. Drupal hooks and Symfony events).

Reply:
OK I made this change